src/dividat-driver/senso/update_firmware.go

```diff
@@ -0,0 +1,48 @@
+package senso
+
+import (
+	"bytes"
+	"context"
+	"encoding/base64"
+	"fmt"
+	"io"
+
+	"github.com/dividat/driver/src/dividat-driver/firmware"
+)
+
+type OnUpdate func(msg FirmwareUpdateMessage)
```

Comment:
I used a channel instead of a callback previously, but ran into issues with some messages not being sent and others being sent twice. I couldn't figure out why that happened so I simplified to use a callback instead.

Reply:
The callback looks good to me :+1: